SwanFileBrowser/src/kernelspec/validate.ts

```diff
@@ -0,0 +1,198 @@
+/**
+ * File took from https://github.com/jupyterlab/jupyterlab/blob/master/packages/services/src/kernelspec/validate.ts
+ * for jupyterlab 3.0.x
+ *
+ * This allows to validate the kernels specs and to get the ISpecModel.
+ * This is required to update the manager.services.kernelspecs.specs object
+ * in the SwanFileBrowserModel.
+ */
```

Comment:
Again here, it would be useful to know what you actually had to change, since it is not obvious what is copied and what has been modified.

Reply:
in this file nothing was modified, the thing is that the function I need from this file is not public.